code/modules/chemistry/tools/ampoules.dm

```diff
@@ -42,6 +42,47 @@
 	playsound(user.loc, "sound/impact_sounds/Generic_Snap_1.ogg", 50, 1)
 	return
 
+// based off of plants.dm
+/obj/item/reagent_containers/ampoule/attackby(obj/item/W as obj, mob/user as mob)
+	if (istype(W, /obj/item/spacecash) || istype(W, /obj/item/paper))
+		var/obj/item/clothing/mask/cigarette/custom/ampoule/C = new(user.loc)
+		C.reagents.maximum_volume = (src.reagents.total_volume)
+		src.force_drop(user)
+		src.set_loc(C)
+		C.ampoule = src
+		src.reagents.copy_to(C.reagents)
```

Comment:
Unless things break when this line is higher up, it should probably be higher up in the code

Reply:
Not sure why it would help if it was higher  up, but I moved it anyway.